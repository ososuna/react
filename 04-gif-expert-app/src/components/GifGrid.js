
import React, { useState, useEffect } from 'react'
import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);
    
    useEffect( async() =>{
        const images = await getGifs( category );
        setImages( images );
    }, [ category ]);
    
    return (
        <>
            <h3>{ category }</h3>
            <div className="card-grid">
                    {
                        images.map( ( img ) => (
                            <GifGridItem
                                key={ img.id }
                                { ...img }
                            />
                        ))
                    }
            </div>
        </>
    )
}
