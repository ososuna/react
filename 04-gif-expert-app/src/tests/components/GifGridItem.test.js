
import React from 'react';
import { shallow } from 'enzyme';

import { GifGridItem } from '../../components/GifGridItem';

describe('GifGridItem component tests', () => {
    
    const title = 'A title';
    const url   = 'https://localhost/something.jpg';
    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );
    
    test('should show <GifGridItem /> correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('should be a paragraph with the title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    });

    test('should be the url and alt of the image', () => {
        
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    
    });
    
    test('should have animate__fadeIn', () => {
        
        const div = wrapper.find('div');
        const className = div.prop('className');
        
        expect( className.includes( 'animate__fadeIn' ) ).toBe( true );

    });
    
});

