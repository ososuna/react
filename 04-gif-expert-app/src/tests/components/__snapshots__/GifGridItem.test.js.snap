// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem component tests should show <GifGridItem /> correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A title"
    src="https://localhost/something.jpg"
  />
  <p>
    A title
  </p>
</div>
`;
